### PR TITLE
Fix transformation bug that may generate multiple duplicate xform results

### PR DESCRIPTION
**Pull request by hsyuan**
CXformSimplifySubquery has 3 transformation subrules, in function FSimplify, if
the subquery is neither quantified subquery nor existential subquery, it will
assign the input expression to the new output expression and return true, which
is wrong and causes the generation of same output expression with the input
expression. The worse thing is that it doubled the size of the search space in
many cases.

Fixes issue #275

### CODE REVIEW

**Comment by hsyuan** [comment]
yes, but I don't know why. The plan doesn't change though.

**Comment by hsyuan** [comment]
According to the figure 3 of section `3.2 Counting Query Plans` in paper `F. Waas, C. Galindo-Legaria, "Counting, Enumerating, and Sampling of Execution Plans in a Cost-Based Query Optimizer", ACM SIGMOD, 2000`
image

By removing the duplicate transform expression (which may smaller cost, suppose we only store top K cheapest alternatives for every group), that memo group is able to store other plan alternatives, these group expressions may have larger number of child alternatives, which causes a larger search space.

File: data/dxl/minidump/Select-Proj-OuterJoin.mdp [hunk 1]
```diff
@@ -1163,7 +1163,7 @@
                   <dxl:IndexCondList>
                     <dxl:Comparison ComparisonOperator="=" OperatorMdid="0.93.1.0">
                       <dxl:Ident ColId="73" ColName="nspname" TypeMdid="0.19.1.0"/>
-                      <dxl:ConstValue TypeMdid="0.19.1.0" IsNull="false" IsByValue="false" Value="cGdfY2F0YWxvZwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&#10;AAAAAAAAAAAAAAAAAAAAAAAAAA=="/>
+                      <dxl:ConstValue TypeMdid="0.19.1.0" IsNull="false" IsByValue="false" Value="cGdfY2F0YWxvZwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&#xA;AAAAAAAAAAAAAAAAAAAAAAAAAA=="/>
```
Comment (hsyuan) [inline data/dxl/minidump/Select-Proj-OuterJoin.mdp]: changed from `tidycat@` to `tidycat`.

Comment (hsyuan) [reply data/dxl/minidump/Select-Proj-OuterJoin.mdp]: changed from `pg_catalog@` to `pg_catalog`

File: data/dxl/minidump/Select-Proj-OuterJoin.mdp [hunk 2]
```diff
@@ -1024,7 +1024,7 @@
                   <dxl:IndexCondList>
                     <dxl:Comparison ComparisonOperator="=" OperatorMdid="0.93.1.0">
                       <dxl:Ident ColId="31" ColName="nspname" TypeMdid="0.19.1.0"/>
-                      <dxl:ConstValue TypeMdid="0.19.1.0" IsNull="false" IsByValue="false" Value="dGlkeWNhdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&#10;AAAAAAAAAAAAAAAAAAAAAAAAAA=="/>
+                      <dxl:ConstValue TypeMdid="0.19.1.0" IsNull="false" IsByValue="false" Value="dGlkeWNhdAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA&#xA;AAAAAAAAAAAAAAAAAAAAAAAAAA=="/>
```
Comment (hsyuan) [inline data/dxl/minidump/Select-Proj-OuterJoin.mdp]: changed from `tidycat@` to `tidycat`.

**Comment by hsyuan** [comment]
@vraghavan78  The changes on the const values are correct. The old value was wrong.

**Comment by vraghavan78** [comment]
Can you change the values in QueryDXL too?

**Comment by hsyuan** [comment]
I changed the const value back to old one, it still passes the test.

**Comment by vraghavan78** [comment]
Thanks for investigating this @hysuan

**Comment by hsyuan** [comment]
see my comment on slack channel.

Sent from my iPhone

**Comment by hsyuan** [comment]
You approved twice. I am gonna count them as two +1s. :)

**Comment by hsyuan** [comment]
This patch was reverted, because it failed on the following test:
```sql
create table A(i integer, j integer) distributed by (i);
insert into A values(1,1);
insert into A values(19,5);
insert into A values(99,62);
insert into A values(1,1);
insert into A values(78,-1);
analyze A;

create table B(i integer, j integer) distributed by (i);
insert into B values(1,43);
insert into B values(88,1);
insert into B values(-1,62);
insert into B values(1,1);
insert into B values(32,5);
insert into B values(2,7);
analyze B;

create table C(i integer, j integer) distributed by (i);
insert into C values(1,889);
insert into C values(288,1);
insert into C values(-1,625);
insert into C values(32,65);
insert into C values(32,62);
insert into C values(3,-1);
insert into C values(99,7);
insert into C values(78,62);
insert into C values(2,7);
analyze C;

select * from A,B where exists (select * from C where C.j = A.j and B.i = all (select min(C.j) from C where C.j = B.j)) order by 1,2,3,4;
  i | j | i | j
 ---+---+---+---
 (0 rows)
```

But the correct results are:
```sql
  i  | j  | i | j
 ----+----+---+---
   1 |  1 | 1 | 1
   1 |  1 | 1 | 1
  78 | -1 | 1 | 1
  99 | 62 | 1 | 1
 (4 rows)
```

**Comment by hsyuan** [comment]
Looks like GPDB master remove this test, that is why it is green on master pipeline, red on 4.3. Need to add it back to 5 & 6. We also need to add this into Orca minidump tests.

**Pull request merged**